src/PSServiceBus/Helpers/SbReceiver.cs

```diff
@@ -167,13 +173,27 @@ public void PurgeMessages()
 
                 do
                 {
-                    res = messageReceiver.ReceiveAsync(100,TimeSpan.FromSeconds(1)).Result;
+                    //Did we invoke the dispose of this receiver?
+                    if (tokenCancel.IsCancellationRequested)
+                    {
+                        break;
+                    }
+
+                    //Is the messageReceiver in a closing state?
+                    if (messageReceiver.IsClosedOrClosing)
+                    {
+                        break;
+                    }
+
+                    var temp = messageReceiver.ReceiveAsync(100, TimeSpan.FromSeconds(1));
```

Comment:
For my own knowledge - what is the difference between these 3 lines and

```
res = messageReceiver.ReceiveAsync(100, TimeSpan.FromSeconds(1)).Result;
```

Reply:
I'm "hoping" that the .Wait() will actually make things wait and not run it in some kind of back ground thread (async) mode.

When reading about the task / async pattern, you will see it is very hard to make synchronous code execute asynchronous code, without things running in the back ground. This is my best guess with my current knowledge, and my tests are showing it works.